File: apps/HSMS-fe/src/app/models/Branch.dto.ts
// src/models/branch.dto.ts
export interface BranchMastersDto {
 id: number;
  companyId: number;
  companyName: string;
  branchName: string;
  branchCode: string;
  email: string;
  phone: string;
  addressLine1: string;
  addressLine2: string | null;
  cityId: number;
  cityName: string;
  stateId: number;
  stateName: string;
  countryId: number;
  countryName: string;
  postalCode: string;
  isMainBranch: boolean;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateBranchDto {
  companyId: number;
  branchName: string;
  branchCode: string;
  email: string;
  phone: string;
  addressLine1: string;
  addressLine2?: string | null;
  cityId: number;
  stateId: number;
  countryId: number;
  postalCode: string;
  isMainBranch: boolean;
  isActive: boolean;
}


export interface UpdateBranchDto {
  id?: number;  
  branchID?: string;  
  branchName: string;
  branchHeader?: string;
  registerName?: string;
  labHeader?: string;
  companyID: number;
  address: string;
  stateID: number;
  districtID: number;
  cityID: number;
  areaID: number;
  mobile1?: string;
  mobile2?: string;
  phone?: string;
  contactPerson?: string;
  
}

File: libs/contracts/Dto/BranchDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HSMS.contracts.Dto
{
    public class BranchDto
    {
        public int Id { get; set; }
        public int CompanyId { get; set; }
        public string? CompanyName { get; set; }
        public string BranchName { get; set; } = string.Empty;
        public string BranchCode { get; set; } = string.Empty;
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string AddressLine1 { get; set; } = string.Empty;
        public string? AddressLine2 { get; set; }
        public int? CityId { get; set; }
        public string? CityName { get; set; }
        public int? StateId { get; set; }
        public string? StateName { get; set; }
        public int? CountryId { get; set; }
        public string? CountryName { get; set; }
        public string? PostalCode { get; set; }
        public bool IsMainBranch { get; set; }
        public bool IsActive { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset UpdatedAt { get; set; }
    }
}
